### PR TITLE
Add Dockerfiles & docker-compose files for official OSS & Plus images

### PR DESCRIPTION
**Pull request by dhurley**
### Proposed changes

Add official Dockerfiles for NGINX OSS & Plus images that have the NGINX Agent installed on them as well.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [x] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [x] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [x] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: scripts/docker/official/nginx-oss-with-nginx-agent/alpine/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+ARG NGINX_VERSION=1.25.1-alpine
+FROM nginx:${NGINX_VERSION}
+LABEL maintainer="NGINX Agent Maintainers <agent@nginx.com>"
+
+RUN set -x \
+    && apk add curl bash iproute2
+
+RUN printf "%s%s\n" \
+    "http://packages.nginx.org/nginx-agent/alpine/v3.16" \
+    "/main" | tee -a /etc/apk/repositories
```
Comment (Dean-Coakley) [inline scripts/docker/official/nginx-oss-with-nginx-agent/alpine/Dockerfile:10]: Makes sense to do something like the following? 

So when `ARG NGINX_VERSION` is modified and includes a alpine version change, the correct agent version will always be installed?

```suggestion
RUN printf "%s%s%s\n" \
    "http://packages.nginx.org/nginx-agent/alpine/v" \
    `grep -o -E '^[0-9]+\.[0-9]+' /etc/alpine-release` \
    "/main" | tee -a /etc/apk/repositories
```

**Review by Dean-Coakley — APPROVED** [review]

File: scripts/docker/official/nginx-plus-with-nginx-agent/alpine/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,68 @@
+FROM alpine:3.17
```
Comment (defanator) [inline scripts/docker/official/nginx-plus-with-nginx-agent/alpine/Dockerfile:1]: Any reason not to use official nginx/foss alpine image as a base here? It should be pretty straightforward and would make future maintenance easier.

Comment (dhurley) [reply scripts/docker/official/nginx-plus-with-nginx-agent/alpine/Dockerfile:1]: This Dockerfile is for NGINX Plus where there isn't an official image. We used the reference here to build an example Dockerfile that people can use if they are using NGINX Plus. https://docs.nginx.com/nginx/admin-guide/installing-nginx/installing-nginx-docker/#creating-nginx-plus-docker-image 

We do use the docker image for NGINX OSS alright here for this other Dockerfile we have https://github.com/nginx/agent/blob/fc8606e8ad47b83a830a77c067d52116d86f7bae/scripts/docker/official/nginx-oss-with-nginx-agent/alpine/Dockerfile

File: scripts/docker/official/nginx-plus-with-nginx-agent/alpine/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        echo "key verification succeeded!"; \
+        mv /tmp/nginx_signing.rsa.pub /etc/apk/keys/; \
+    else \
+        echo "key verification failed!"; \
+        exit 1; \
+    fi \
+    && cat cert.pem > /etc/apk/cert.pem \
+    && cat cert.key > /etc/apk/cert.key \
+    && apk add -X "https://pkgs.nginx.com/plus/alpine/v$(egrep -o '^[0-9]+\.[0-9]+' /etc/alpine-release)/main" --no-cache $nginxPackages \
+    && apk add -X "https://pkgs.nginx.com/nginx-agent/alpine/v3.16/main" --no-cache nginx-agent \
```
[the remark this answers is not in the document]
Comment (dhurley) [reply scripts/docker/official/nginx-plus-with-nginx-agent/alpine/Dockerfile:48]: Yeah at the moment v3.17 is missing for the nginx-agent. Once that is made available in pkgs.nginx.com, I'll update the Dockerfile.

File: scripts/docker/official/nginx-plus-with-nginx-agent/alpine/Dockerfile (hunk 2 above)
Comment (defanator) [reply scripts/docker/official/nginx-plus-with-nginx-agent/alpine/Dockerfile:1]: I'm trying to say that the only difference between nginx/foss image and N+ image is nginx package itself; it's much easier to take the same base image (foss one) and just replace the package from nginx to nginx-plus. That way you won't need to keep and maintain separate Dockerfile. YMMV.

Comment (thresheek) [reply scripts/docker/official/nginx-plus-with-nginx-agent/alpine/Dockerfile:1]: The size of the resulting image would probably be larger if the package is replaced.

**Review by craigell — APPROVED** [review]

File: scripts/docker/official/nginx-plus-with-nginx-agent/alpine/Dockerfile (hunk 2 above)
Comment (dhurley) [reply scripts/docker/official/nginx-plus-with-nginx-agent/alpine/Dockerfile:1]: Yes the size of the image would be bigger. Also there isn't mush difference in terms of number of lines in the Dockerfile with either approach. I would prefer to follow the dockerfile provided by nginx-plus to be consistent.

**Review by aphralG — APPROVED** [review]

**Review by oliveromahony — APPROVED** [review]

**Pull request merged**